### DISCUSSION TITLE
Date picker

### DISCUSSION THREAD

**Discussion by fredericoregateiro** [quasarframework/quasar#10056] [General - Components / Directives / etc]
**Describe the bug**
Show a date popup from a dialog.
I really don't know what i'm doing wrong.

The problem is when i open the dialog on this [line](https://github.com/SolRIA/atlanticgrids/blob/b9af53f8b04596dc2c93198a6b91bcf063cf04eb/src/pages/Registed/Projects.vue#L103) and click on the input icon with the date field, to open the date picker.

**Codepen/jsFiddle/Codesandbox (required)**
[Sample](https://github.com/SolRIA/atlanticgrids)

**To Reproduce**
1. Create a dialog with a date picker inside a input control like this [example](https://quasar.dev/vue-components/date#with-qinput)
2. Click on the icon to open the date control
3. On the PC nothing hapens
4. On the mobile this error on the console: 
```
innerRef.value.click is not a function 
handleShow QDialog.js:217
```

**Expected behavior**
Open the date picker with no error

**Platform (please complete the following information):**
Quasar Version: 2.0.2
@quasar/app Version: v3.0.2

Quasar mode:
  - [x] SPA
  
Tested on:
  - [x] SPA

OS:Win10
Browsers: Edge, Firefox

**Additional context**

**Comment by ontwikkelfabriek** [comment]
@fredericoregateiro 

This is the discussion forum about Quasar related subjects. If you think you have found a **bug** then please create a github **issue**( instead of a discussion).

**Reply by hawkeye64** [reply]
@ontwikkelfabriek This looks like it was an issue converted to a discussion because it's using the issue template.

**Comment by fredericoregateiro** [comment]
I don't know if it's really a bug, but I can't see what's going on.
Can anyone help me see what the problem is?

**Reply by metalsadman** [reply]
@fredericoregateiro where is this code found in your repo?
```js
innerRef.value.click is not a function 
handleShow QDialog.js:217
```
you can link the file and line, would be easier for peeps to help you.

**Reply by fredericoregateiro** [reply]
The problem is when i open the dialog on this [line](https://github.com/SolRIA/atlanticgrids/blob/b9af53f8b04596dc2c93198a6b91bcf063cf04eb/src/pages/Registed/Projects.vue#L103) and click on the input icon with the date field, to open the date picker.

**Reply by fredericoregateiro** [reply]
Can anyone help me on this? I don't know what's going on.

**Comment by pdanpdan** [comment]
Something is very strange, because that block of code (in QDialog) only runs on iOS, and you mentioned:

OS:Win10
Browsers: Edge, Firefox

**Reply by fredericoregateiro** [reply]
Hello @pdanpdan thanks for reply

I had only tested it on Windows when I opened this thread, but I just tried it on an ipad and it doesn't work either.

In the [documentation](https://quasar.dev/quasar-plugins/dialog#invoking-custom-component) I'm following there is no reference to only work on iOS

**Comment by pdanpdan** [comment]
The code that you linked in the OP is only run on iOS, so it could not happen to you on windows.
Unfortunately I cannot reproduce the problem. I'll convert it back to issue and maybe someone will be able to reproduce it.

**Reply by fredericoregateiro** [reply]
@pdanpdan sorry i didn't understand what you first said.

The error
```
innerRef.value.click is not a function 
handleShow QDialog.js:217
```

i get using the iphone simulator, using the devtools in chrome or edge, and only using this simulator, any android device or in the web, nothing happens and the datepicker doesn't show

![image](https://user-images.githubusercontent.com/6304701/127352687-eba4f943-28d6-40b3-bb75-022460405e9e.png)


I've uploaded a build version on this [site](https://atlanticgrids.solria.pt/#/registed)

**Comment by exninja** [comment]
Try changing the icon to a constant instead of injecting the SVG icon.
I think there's a bug with Quasar v2 with the "QDate with QInput" [demo](https://quasar.dev/vue-components/date#with-qinput) when using a variable on the icon attribute.

e.g.
Change :

<q-btn :icon="mdiCalendarMonth"...

to :

<q-btn icon="mdi-calendar-month"...

(or whatever non-SVG icon you can use for now)

**Reply by fredericoregateiro** [reply]
I'm using scg's so i don't have to import the font, just to use some icons.

I've found that the problem is with the icon and svg usage, and for me there is a better work around [codepen sample](https://codepen.io/regateiro/pen/jOmKaEY)

**Reply by exninja** [reply]
Nice.
